pkg/nfd-worker/nfd-worker.go

```diff
@@ -190,12 +190,16 @@ func (i *infiniteTicker) Reset(d time.Duration) {
 
 // Run feature discovery.
 func (w *nfdWorker) runFeatureDiscovery() error {
+	durations := make(map[string]string, len(w.featureSources))
 	for _, s := range w.featureSources {
 		klog.V(2).Infof("running discovery for %q source", s.Name())
+		discoverStart := time.Now()
 		if err := s.Discover(); err != nil {
 			klog.Errorf("feature discovery of %q source failed: %v", s.Name(), err)
 		}
+		durations[s.Name()] = time.Since(discoverStart).String()
 	}
+	utils.KlogDump(3, "discovery durations for feature sources:", "  ", durations)
```

Comment:
What's the advantage of using intermediate map & printing things only after whole loop, instead of just printing header before loop, moving Infof() printing after Discover(), and adding time to that log item?

Reply:
I removed the `Infof` before `Discover` and added logging after which includes the duration.